File: lab_2/task1/rand_sequence.cpp
#include <iostream>
#include <random>
#include <sstream>

const int SIZE = 128;
using namespace std;

/**
 * Generates a random binary sequence of a specified size.
 *
 * @return A string containing a random binary sequence.
 */
 
string generateRandomSequence() {
    string sequence;
    random_device rd;
    mt19937 gen(rd());
    uniform_int_distribution<> dis(0, 1);

    for (int i = 0; i < SIZE; ++i) {
        stringstream ss;
        ss << dis(gen);
        sequence += ss.str();
    }

    return sequence;
}

int main() {
    string randomSequence = generateRandomSequence();
    cout << "Generated Sequence: " << randomSequence << endl;

    return 0;
}

File: lab_2/task1/rand_sequence.java
import java.util.Random;

/**
 * Utility class for generating random binary sequences.
 */ 
public class RandomSequenceGenerator {
    /**
     * Generates a random binary sequence of length 128.
     *
     * @return A string containing a random binary sequence.
     */
    public static String generateRandomSequence() {
        StringBuilder sequence = new StringBuilder();
        Random random = new Random();

        for (int i = 0; i < 128; ++i) {
            sequence.append(random.nextInt(2));
        }

        return sequence.toString();
    }
    
    /**
     * Main method to demonstrate the usage of the generateRandomSequence method.
     *
     * @param args Command-line arguments (not used).
     */
    public static void main(String[] args) {
        String randomSequence = generateRandomSequence();
        System.out.println("Generated Sequence: " + randomSequence);
    }
}